addons/event_sale/models/sale_order_line.py

```diff
@@ -70,6 +76,22 @@ def _compute_name(self):
         """
         super()._compute_name()
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        """ Don't create sale order line that needs an event and event ticket. We do this to avoid
+        raising the error linked to the constrain as we don't want to block the user flow when he
+        clicks on "Configure Event" to create his first event. """
+        so_lines_to_keep = []
+        for values in vals_list:
+            product = self.env['product.product'].browse(values.get('product_id'))
+            if product and product.detailed_type == 'event':
+                if values.get('event_ticket_id') and values.get('event_id'):
+                    so_lines_to_keep.append(values)
+            else:
+                so_lines_to_keep.append(values)
+
+        return super().create(so_lines_to_keep)
```

Comment:
To add up on this, I just tested something, if you create a registration, then manually empty the "ticket type", then confirm the SO, it removes your sale order line, which is very strange ^^